pkg/bucket/policy/policy.go

```diff
@@ -100,27 +100,20 @@ func (policy Policy) MarshalJSON() ([]byte, error) {
 	return json.Marshal(subPolicy(policy))
 }
 
+// MergeStatements adds new statements to the existing policy
+// and drop duplicate statements if any.
+func (policy *Policy) MergeStatements(statements []Statement) {
```

Comment:
I think it should be called merged Policies @vadmeste - Statements do not exist outside of a policy - internally it merges the statements anyways. 

```
```suggestion
func (policy *Policy) MergePolicy(mpolicy Policy) {
```

Reply:
Also, this function should return a new policy as a clone() - we shouldn't modify the `*Policy`  using them concurrently is unsafe and there are several places they may get used concurrently. 

The Merge() should return a Policy.